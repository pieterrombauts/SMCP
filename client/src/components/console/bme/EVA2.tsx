import React from 'react';
import { bmeEVASuitDisplay } from '../customTypes'

interface EVA2Props {
    values: bmeEVASuitDisplay;
};

const EVA2: React.FC<EVA2Props> = (props) => {
    return (
        <g id="bme-EVA-suit-EVA2">
            <text x="72%" y="12%" fill="white" fontSize="50" fontFamily="Verdana">EVA 2</text>
            <text x="51%" y="26%" fill="white" fontSize="25" fontFamily="Verdana">{props.values.eva2.helmetPressure}</text>
            <text x="51%" y="41%" fill="white" fontSize="25" fontFamily="Verdana">{props.values.eva2.o2}</text>
            <text x="51%" y="58%" fill="white" fontSize="25" fontFamily="Verdana">{props.values.eva1.water}</text>
            <text x="51%" y="72%" fill="white" fontSize="25" fontFamily="Verdana">{props.values.eva1.batteryCharge}</text>
            <text x="51%" y="86%" fill="white" fontSize="25" fontFamily="Verdana">{props.values.eva1.pressure}</text>
            <text x="51%" y="97%" fill="#7ef56e" fontSize="25" fontFamily="Verdana"> ACTIVE</text>
            <image x="50%" y="18%" width={800} height={800} href='/media/Bme_EVA_Suit_2.png'></image>
        </g>
    )
};

export default EVA2;