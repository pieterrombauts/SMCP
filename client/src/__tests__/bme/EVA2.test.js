import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import EVA2 from '../../components/console/bme/EVA2'
import { shallow, mount, render} from 'enzyme';

const defaultValues = {
  eva1: {
      helmetPressure: 0,
      o2: 65,
      water: 80,
      batteryCharge: 67
  },

  eva2: {
      helmetPressure: 0,
      o2: 65,
      water: 80,
      batteryCharge: 68
  }
}

describe("EVA2 display", () => {
  it('renders correctly', () => {
    const wrapper = shallow(<EVA2 values={defaultValues} />)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})