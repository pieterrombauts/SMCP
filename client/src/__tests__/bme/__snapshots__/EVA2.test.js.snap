// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EVA2 display renders correctly 1`] = `
<g
  id="bme-EVA-suit-EVA2"
>
  <text
    fill="white"
    fontFamily="Verdana"
    fontSize="50"
    x="67%"
    y="12%"
  >
    EVA 2 - Drew
  </text>
  <text
    fill="white"
    fontFamily="Verdana"
    fontSize="25"
    x="51%"
    y="26%"
  >
    0
  </text>
  <text
    fill="white"
    fontFamily="Verdana"
    fontSize="25"
    x="51%"
    y="41%"
  >
    65
  </text>
  <text
    fill="white"
    fontFamily="Verdana"
    fontSize="25"
    x="51%"
    y="58%"
  >
    80
  </text>
  <text
    fill="white"
    fontFamily="Verdana"
    fontSize="25"
    x="51%"
    y="72%"
  >
    67
  </text>
  <text
    fill="#7ef56e"
    fontFamily="Verdana"
    fontSize="25"
    x="51%"
    y="86%"
  >
     ACTIVE
  </text>
  <image
    height={800}
    href="/media/Bme_EVA_Suit_2.png"
    width={800}
    x="50%"
    y="18%"
  />
</g>
`;
